connector/connector.go

```diff
@@ -61,383 +52,100 @@ type Metrics interface {
-	component.Component
-	consumer.Logs
-}
+type Logs = internal.Logs
 
 // CreateSettings configures Connector creators.
 //
 // Deprecated: [v0.103.0] Use connector.Settings instead.
-type CreateSettings = Settings
+type CreateSettings = internal.Settings
 
 // Settings configures Connector creators.
-type Settings struct {
-	// ID returns the ID of the component that will be created.
-	ID component.ID
-
-	component.TelemetrySettings
-
-	// BuildInfo can be used by components for informational purposes
-	BuildInfo component.BuildInfo
-}
+type Settings = internal.Settings
 
 // Factory is factory interface for connectors.
 //
 // This interface cannot be directly implemented. Implementations must
 // use the NewFactory to implement it.
-type Factory interface {
-	component.Factory
-
-	// CreateDefaultConfig creates the default configuration for the Connector.
-	// This method can be called multiple times depending on the pipeline
-	// configuration and should not cause side-effects that prevent the creation
-	// of multiple instances of the Connector.
-	// The object returned by this method needs to pass the checks implemented by
-	// 'configtest.CheckConfigStruct'. It is recommended to have these checks in the
-	// tests of any implementation of the Factory interface.
-	CreateDefaultConfig() component.Config
-
-	CreateTracesToTraces(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Traces) (Traces, error)
-	CreateTracesToMetrics(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Metrics) (Traces, error)
-	CreateTracesToLogs(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Logs) (Traces, error)
-
-	CreateMetricsToTraces(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Traces) (Metrics, error)
-	CreateMetricsToMetrics(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Metrics) (Metrics, error)
-	CreateMetricsToLogs(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Logs) (Metrics, error)
-
-	CreateLogsToTraces(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Traces) (Logs, error)
-	CreateLogsToMetrics(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Metrics) (Logs, error)
-	CreateLogsToLogs(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Logs) (Logs, error)
-
-	TracesToTracesStability() component.StabilityLevel
-	TracesToMetricsStability() component.StabilityLevel
-	TracesToLogsStability() component.StabilityLevel
-
-	MetricsToTracesStability() component.StabilityLevel
-	MetricsToMetricsStability() component.StabilityLevel
-	MetricsToLogsStability() component.StabilityLevel
-
-	LogsToTracesStability() component.StabilityLevel
-	LogsToMetricsStability() component.StabilityLevel
-	LogsToLogsStability() component.StabilityLevel
-
-	unexportedFactoryFunc()
-}
+type Factory = internal.Internal
```

Comment:
Not blocking as this is internal but why is this not called `internal.Factory`?

Reply:
I followed what @mx-psi did in this PR: https://github.com/open-telemetry/opentelemetry-collector/pull/10253/files#diff-56869497b872826bc1bd89731571d212be40aa7fe7252864a80317e6413234e3R126

Since there is a private `factory` struct, I guess it makes sense to use an entirely different name to avoid mixing them.

Reply:
I am fine with `internal.Factory`, there was no real reason for using `internal.Internal`

Reply:
I have renamed `internal.Internal` to `internal.Factory`.